File: Semester 2/Class03_Computer_Contest_Lv_2_Notes_and_Homework_Jan_29/lvl2_hw_sol/lvl2_fujo_neko.cpp
#include <bits/stdc++.h>
using namespace std;
int R, C, Q; bool row[1003], col[1003]; char g[1003];
int main(){
    scanf("%d %d", &R, &C);
    for(int i=1; i<=R; i++){
        scanf("%s", g+1);
        for(int j=1; j<=C; j++)
            if(g[j]=='X') row[i]=1, col[j]=1;
    }
    scanf("%d", &Q);
    for(int i=0, x, y; i<Q; i++){
        scanf("%d %d", &x, &y);
        if(row[y] || col[x]) printf("Y\n");
        else printf("N\n");
    }
}

File: Semester 2/Class03_Computer_Contest_Lv_2_Notes_and_Homework_Jan_29/lvl2_hw_sol/lvl2_fujo_neko.java
import java.util.*;
import java.io.*;
public class homework {
	static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	static PrintWriter pr = new PrintWriter(new BufferedWriter(new OutputStreamWriter(System.out)));
	static StringTokenizer st;
	public static void main(String[] args) throws IOException{
		int R = readInt(), C = readInt(); 
		boolean [] row = new boolean[R], col = new boolean[C];
		for(int i=0; i<R; i++) {
			String s = readLine();
			for(int j=0; j<C; j++) {
				if(s.charAt(j) == 'X') { 
					row[i]=true; col[j] = true; 
				}
			}
		}
		for(int q = readInt(); q > 0 ; q--) {
			int c = readInt() - 1, r = readInt() - 1;
			if(row[r] || col[c]) System.out.println("Y");
			else System.out.println("N");
		}
	}
	static String next () throws IOException {
		while (st == null || !st.hasMoreTokens())
			st = new StringTokenizer(br.readLine().trim());
		return st.nextToken();
	}
	static long readLong () throws IOException {
		return Long.parseLong(next());
	}
	static int readInt () throws IOException {
		return Integer.parseInt(next());
	}
	static double readDouble () throws IOException {
		return Double.parseDouble(next());
	}
	static char readCharacter () throws IOException {
		return next().charAt(0);
	}
	static String readLine () throws IOException {
		return br.readLine().trim();
	}
}